### ISSUE TITLE
Legacy API versions remain after upgrading

### ISSUE THREAD

**Issue by sjames-hms** [LawrenceLoz/FormulaShare-DX#129]
My org was originally using a legacy version of FormulaShare and in our sandbox, I recently upgraded to version 3.27. However, the following class and test class: [fflib_InOrder] and [fflib_InOrderTest] have an API version of 30. We are trying to confirm that all of the legacy API versions from 21.0 to 30.0 are removed from our org. Is there another upgrade of FormulaShare we should be using? Thank you.

**Comment by LawrenceLoz** [comment]
Hi @sjames-hms many thanks for flagging this.

3.27 is the latest version of the FormulaShare managed package, though the class you mention and several others from the fflib libraries we use in the app are on some quite old API versions.

These are kept up to date in the project but it appears there's been an issue with the version updates for libraries getting through to our packaging org. We'll certainly resolve this in our Winter release which we're aiming to distribute for use in production around December/January and ensure all classes are on recent versions.

If the work you're doing to remove legacy API versions relates to the upcoming retirement of versions 21.0 to 30.0 it's worth noting this doesn't directly affect versioned metadata components within a package (https://help.salesforce.com/s/articleView?id=000389618&type=1).

I can appreciate from a governance perspective though you may need to ensure outdated API versions aren't used across the board. Let me know if it waiting until December/January would be an issue for you - we'd be happy to fast-track a new package version with the API version updates in place if needed.

I'll keep this issue open until this is resolved in any case

**Comment by sjames-hms** [comment]
Yes, we can wait until Dec/Jan for the updated package. Thank you.

**Comment by LawrenceLoz** [comment]
Hi @sjames-hms, yes we're working on the final changes and testing of the new version at the moment - this is coming a little later than I'd shared as we wanted to make use of a feature available in Spring '25.

The API version changes have already been made so these will definitely be in place when the version comes, but apologies I know this is a little later than the Dec/Jan timeline I'd mentioned before. We should have the version available for package upgrades late this month.

Anything you'd like to check in the meantime just let me know

**Comment by sjames-hms** [comment]
Okay thank you.

**Comment by LawrenceLoz** [comment]
@sjames-hms you may have seen this in our release email recently, but to confirm the new FormulaShare package version which includes updated API classes is now available. You can either upgrade when you're ready from the AppExchange: https://appexchange.salesforce.com/appxListingDetail?listingId=a0N3A00000FR5TCUA1, or wait for the automatic sandbox and production upgrades.

Info on new features and upgrade schedule: https://cloudsundial.com/formulashare-spring-25

**Comment by sjames-hms** [comment]
Okay great, we will wait for the automatic upgrade @LawrenceLoz 

Is there a listserv that we need to be on to be notified when the upgrades are complete?

**Comment by LawrenceLoz** [comment]
Hi @sjames-hms, we'll email again when all production org upgrades are complete. If you received an email on 17th/18th Feb you'll receive one again at this point. If not and you'd like to get release notifications then drop us an email at formulashare@cloudsundial.com and we'll add you to the list!
